dali/python/nvidia/dali/external_source.py

```diff
@@ -664,18 +664,33 @@ def external_source(source = None, num_outputs = None, *, cycle = None, name = N
     which is more performant.
     """
 
-    if batch is None:
-        batch = True
+    from nvidia.dali._debug_mode import _PipelineDebug
+
+    def _external_source(source = None, num_outputs = None, *, cycle = None, name = None, device = "cpu", layout = None,
+                    cuda_stream = None, use_copy_kernel = None, batch = True, **kwargs):
+        if batch is None:
+            batch = True
+
+        if num_outputs is not None:
+            if source is None:
+                raise ValueError("The parameter ``num_outputs`` is only valid when using ``source`` to "
+                    "provide data. To feed multiple external sources in ``feed_input``, use multiple "
+                    "``external_source`` nodes.")
+
+        op = ExternalSource(device = device, num_outputs = num_outputs, source = source,
+                            cycle = cycle, layout = layout, cuda_stream = cuda_stream,
+                            use_copy_kernel = use_copy_kernel, batch = batch, **kwargs)
+        return op(name = name)
+
+    # Wrapper around external_source to switch between standard and debug mode.
+    current_pipeline = _PipelineDebug.current()
+    if getattr(current_pipeline, '_debug_on', False):
+        return current_pipeline._external_source(_external_source, source, num_outputs, cycle=cycle, name=name,
```

Comment:
Here we have underscore-prefixed method outside of its class. That means that either the name is incorrect or it shouldn't be used here. I'd keep the convention of having `_foo` private within class.

Reply:
I think the intention is to keep them private for the library use and just hide them from the user.

Reply:
Yes, for it to work, we would have to move all of the underscored methods outside of the `PipelineDebug` or expose it to the "public" API.